# Kuis 1 — pandas dasar (Pertemuan 2)
**Waktu: 20 menit · Individu · Bobot: 3% dari nilai akhir**

**Aturan:**
- Boleh membuka catatan, notebook praktikum, dan dokumentasi pandas.
- **Tidak boleh** menggunakan asisten AI, dan tidak boleh berdiskusi dengan mahasiswa lain.
- Ganti nama file menjadi `P02_Kuis1_<NIM>_<Nama>.ipynb`.
- Kumpulkan: *Restart & Run all* → pastikan tanpa error → unduh `.ipynb` → unggah ke LMS.

**Data:** `kunjungan_perpustakaan_2025.csv` — catatan kunjungan perpustakaan kampus, Februari–Juni 2025.
Satu baris = satu kunjungan.

In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv('../data/kunjungan_perpustakaan_2025.csv')
df.head()

,id_kunjungan,tanggal,nim_hash,prodi,angkatan,durasi_menit,jumlah_pinjam,jenis_ruang,kepuasan
0,K100000,2025-05-18,H1160,Sistem Informasi,2024,30,0,Lab Komputer,5.0
1,K100001,2025-02-22,H1190,Teknik Informatika,2023,78,1,Ruang Baca,3.6
2,K100002,2025-06-11,H1070,Sistem Informasi,2025,56,2,Ruang Baca,4.1
3,K100003,2025-05-06,H1001,Sistem Informasi,2024,12,1,Lab Komputer,2.6
4,K100004,2025-02-24,H1047,Teknik Informatika,2023,67,1,Lab Komputer,4.3


---
### Tugas 1 (15 poin) — Memuat dan memeriksa

In [10]:
# a. Cetak jumlah baris dan kolom
print(f"Jumlah baris: {df.shape[0]}, Jumlah kolom: {df.shape[1]}")
# b. Cetak jumlah nilai kosong (missing) untuk setiap kolom
print("\nJumlah missing value per kolom:")
print(df.isnull().sum())
# c. Cetak berapa mahasiswa unik (nim_hash) yang pernah berkunjung
print(f"\nJumlah mahasiswa unik: {df['nim_hash'].nunique()}")


Jumlah baris: 900, Jumlah kolom: 9

Jumlah missing value per kolom:
id_kunjungan      0
tanggal           0
nim_hash          0
prodi             0
angkatan          0
durasi_menit      0
jumlah_pinjam     0
jenis_ruang       0
kepuasan         45
dtype: int64

Jumlah mahasiswa unik: 198


---
### Tugas 2 (20 poin) — Filter

In [12]:
# Ambil kunjungan mahasiswa prodi 'Teknik Informatika' dengan durasi_menit > 120.
fil = df[(df['prodi'] == 'Teknik Informatika') & (df['durasi_menit'] > 120)]

# Cetak jumlah barisnya dan rata-rata jumlah_pinjam pada kelompok tersebut (2 desimal).
print(f"Jumlah baris: {len(fil)}")
print(f"Rata-rata jumlah pinjam: {fil['jumlah_pinjam'].mean():.2f}")


Jumlah baris: 50
Rata-rata jumlah pinjam: 1.02


---
### Tugas 3 (20 poin) — Kolom turunan

In [13]:
# a. Buat kolom 'durasi_jam' = durasi_menit / 60, dibulatkan 2 desimal
df['durasi_jam'] = (df['durasi_menit'] / 60).round(2)

# b. Buat kolom 'lama' berisi 'Panjang' jika durasi_menit >= 90, selain itu 'Singkat'
df['lama'] = np.where(df['durasi_menit'] >= 90, 'Panjang', 'Singkat')

# c. Tampilkan 5 kunjungan dengan durasi terlama (kolom: prodi, jenis_ruang, durasi_jam)
top5 = df.sort_values(by='durasi_menit', ascending=False)[['prodi', 'jenis_ruang', 'durasi_jam']].head(5)
top5


,prodi,jenis_ruang,durasi_jam
283,Bisnis Digital,Ruang Baca,3.07
533,Teknik Informatika,Ruang Diskusi,2.95
749,Teknik Informatika,Ruang Diskusi,2.82
756,Teknik Informatika,Ruang Baca,2.78
542,Teknik Informatika,Ruang Baca,2.77


---
### Tugas 4 (25 poin) — Agregasi

In [14]:
# Buat tabel ringkasan per 'jenis_ruang' yang memuat:
#   - jumlah kunjungan
#   - rata-rata durasi_menit (1 desimal)
#   - total jumlah_pinjam
#   - rata-rata kepuasan (2 desimal)
# Urutkan dari rata-rata durasi terbesar. Simpan ke variabel 'ringkas' dan tampilkan.


In [17]:
ringkasan = df.groupby('jenis_ruang').agg(
    jumlah_kunjungan=('id_kunjungan', 'count'),
    rata_rata_durasi=('durasi_menit', lambda x: round(x.mean(), 1)),
    total_jumlah_pinjam=('jumlah_pinjam', 'sum'),
    rata_rata_kepuasan=('kepuasan', lambda x: round(x.mean(), 2))
).reset_index()
ringkasan = ringkasan.sort_values(by='rata_rata_durasi', ascending=False)
ringkasan

,jenis_ruang,jumlah_kunjungan,rata_rata_durasi,total_jumlah_pinjam,rata_rata_kepuasan
1,Ruang Baca,405,77.2,452,4.00
0,Lab Komputer,184,73.4,203,3.96
2,Ruang Diskusi,311,72.9,338,3.97


---
### Tugas 5 (10 poin) — Menyimpan hasil

In [18]:
# Simpan tabel 'ringkas' ke file 'ringkasan_ruang.csv'
# Simpan tabel 'ringkas' ke file 'ringkasan_ruang.csv'
ringkasan.to_csv('ringkasan_ruang.csv', index=False)


---
### Tugas 6 (10 poin) — Interpretasi
Tulis **3–4 kalimat** berdasarkan tabel Tugas 4:
1. Apa pola yang Anda lihat?
2. Apa SATU hal yang belum bisa Anda simpulkan dari data ini, dan mengapa?

Berdasarkan tabel ringkasan yang dibuat, terlihat pola bahwa Ruang Baca merupakan ruangan yang paling populer (kunjungan terbanyak: 405 kali)dengan durasi kunjungan terlama (rata-rata 77,2 menit), dan menghasilkan peminjaman buku terbanyak (452 buku), dengan tingkat kepuasan tertinggi (4,00).Rata-rata kepuasan di ketiga jenis ruang berada pada rentang yang sangat mirip, yaitu sekitar 3,96 hingga 4,00.

tetapi kita belum bisa menyimpulkan apakah durasi kunjungan yang lebih lama secara langsung menyebabkan mahasiswa meminjam lebih banyak buku atau merasa lebih puas. Hal ini karena data agregasi ini hanya memperlihatkan ringkasan rata-rata dan belum menguji hubungan sebab-akibat (kausalitas) maupun korelasi di tingkat kunjungan individu.